src/NLog/Targets/FileTarget.cs

```diff
@@ -108,12 +101,13 @@ public FileTarget()
             this.OpenFileCacheTimeout = -1;
             this.OpenFileCacheSize = 5;
             this.CreateDirs = true;
-            this.fileArchive = new DynamicFileArchive(MaxArchiveFiles);
+            this.fileArchive = new DynamicFileArchive(0);
```

Comment:
The code has the exactly same outcome, in my understanding, but I agree with you @304NotModified it can be confusing. 

The property setter of MaxArchiveFiles in FileTarget sets directly the Size property of DynamicFileArchive. MaxArchiveFiles has as default value 0. This is because it was passed to the constructor. 

As passing 0 value in the constructor of DynamicFileArchive was confusing I removed the parametrised constructor completely.